pyquil/analytics.py

```diff
@@ -0,0 +1,310 @@
+#!/usr/bin/python
+##############################################################################
+# Copyright 2016-2017 Rigetti Computing
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+##############################################################################
+"""
+A collection of methods for analyzing pyquil programs (gate count, etc...)
+"""
+
+import pyquil.gates as gates
```

Comment:
Dependencies in PEP-8 are split into three categories.  For reference see:
https://www.python.org/dev/peps/pep-0008/#imports